src/lib.rs

```diff
@@ -259,62 +268,80 @@ impl FuzzyFinder {
     /// let finder = FuzzyFinder::new();
     /// println!("{:?}", finder.data_version());
     /// ```
+    #[must_use]
     pub fn data_version(&self) -> &str {
-        return &self.data_version;
+        &self.data_version
     }
 }
 
-/// It's most recommend to use, combine both [Finder] and [FuzzyFinder],
-/// if [FuzzyFinder] got no data, then use [Finder].
+impl Default for FuzzyFinder {
+    /// ```rust
+    /// use tzf_rs::FuzzyFinder;
+    ///
+    /// let finder = FuzzyFinder::new();
+    /// ```
+    fn default() -> Self {
+        let file_bytes: Vec<u8> = load_preindex();
+        Self::from_pb(gen::PreindexTimezones::try_from(file_bytes).unwrap_or_default())
+    }
+}
+
+/// It's most recommend to use, combine both [`Finder`] and [`FuzzyFinder`],
+/// if [`FuzzyFinder`] got no data, then use [`Finder`].
 pub struct DefaultFinder {
     finder: Finder,
     fuzzy_finder: FuzzyFinder,
 }
 
-impl DefaultFinder {
+impl Default for DefaultFinder {
     /// ```rust
     /// use tzf_rs::DefaultFinder;
     /// let finder = DefaultFinder::new();
     /// ```
-    pub fn new() -> DefaultFinder {
-        let finder = Finder::new();
-        let fuzzy_finder = FuzzyFinder::new();
-        let df = DefaultFinder {
+    fn default() -> Self {
+        let finder = Finder::default();
+        let fuzzy_finder = FuzzyFinder::default();
+
+        Self {
             finder,
             fuzzy_finder,
-        };
-        return df;
+        }
     }
+}
 
+impl DefaultFinder {
     /// ```rust
     /// use tzf_rs::DefaultFinder;
     /// let finder = DefaultFinder::new();
     /// assert_eq!("Asia/Shanghai", finder.get_tz_name(116.3883, 39.9289));
     /// ```
+    #[must_use]
     pub fn get_tz_name(&self, lng: f64, lat: f64) -> &str {
         let fuzzy_name = self.fuzzy_finder.get_tz_name(lng, lat);
-        if fuzzy_name != "" {
+        if !fuzzy_name.is_empty() {
             return fuzzy_name;
         }
-        return self.finder.get_tz_name(lng, lat);
+        self.finder.get_tz_name(lng, lat)
     }
 
     /// ```rust
     /// use tzf_rs::DefaultFinder;
     /// let finder = DefaultFinder::new();
```

Comment:
Seems all `new` should be replaced by `default`

Reply:
Arguably. Or one could add a "new()" function that just calls the "default()" function.

Reply:
It also breaks test cases and need update to make CI pass.

Reply:
I won't be able to get to it until the weekend, but I'll fix it then.